mirascope/core/openai/openai_call.py

```diff
@@ -42,6 +42,17 @@ def decorator(fn: Callable[P, R]) -> Callable[P, OpenAICallResponse]:
         def call(*args: P.args, **kwargs: P.kwargs) -> OpenAICallResponse:
             prompt_template = inspect.getdoc(fn)
             assert prompt_template is not None, "The function must have a docstring."
+
+            # Try to get the dictionary for tools from the function result
+            # tools = []
+            fn_result = fn(*args, **kwargs)
+            if isinstance(fn_result, dict):
+                if fn_result_tools := fn_result.get("tools"):
+                    for fn_result_tool in fn_result_tools:
+                        # TODO: Generate tools
+                        # tools.append(generate(fn_result_tool))
+                        pass
```

Comment:
@willbakst 
Do you have any thoughts on the design and implementation policy for this part of the to generate `tools`? On an abstract level, I have looked through this page of the documentation.
https://docs.mirascope.io/latest/concepts/tools_%28function_calling%29/

Reply:
OK, I confirmed that you implemneted it. Thank  you.